dapps/assets/src/components/contacts/Contacts.ts

```diff
@@ -0,0 +1,88 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+// vue imports
+import Component, { mixins } from 'vue-class-component';
+
+// evan.network imports
+import { EvanComponent } from '@evan.network/ui-vue-core';
+import { ContactsService, Contact } from './ContactsService';
+
+@Component
+export default class ContactsComponent extends mixins(EvanComponent) {
+  contactService: ContactsService;
+
+  isLoading = true;
+
+  sortBy = 'name';
+
+  filter = null;
+  filterBy: string[] = [];
+
+  data: Contact[] = [];
+
+  columns = [
+    { key: 'icon', label: '', sortable: false },
+    { key: 'alias', label: this.$t('_assets.contacts.name'), sortable: true },
+    { key: 'updatedAt', label: this.$t('_assets.contacts.updated'), sortable: true },
+    { key: 'createdAt', label: this.$t('_assets.contacts.created'), sortable: true },
+    { key: 'favorite', label: '', sortable: false }
+  ];
+
+  created() {
+    const runtime = this.getRuntime();
+    this.contactService = new ContactsService(runtime);
+  }
+
+  async mounted() {
+    this.data = await this.fetchContacts();    
+    this.isLoading = false;
+  }
+
+  handleRowClicked(contact: Contact) {
+    window.location.hash = `/${this.dapp.rootEns}/profile.vue.${this.dapp.domainName}/${contact.address}/detail`;
+  }
+
+  async handleContactAdded() {
+    console.log('Contact Added');
+    
+    this.isLoading = true;
+    this.data = await this.fetchContacts();
+    this.isLoading = false;
+  }
+
+  async fetchContacts(): Promise<Contact[]> {
+    return this.contactService.getContacts();
+  }
+
+  filterByType(type: string) {
+    this.filterBy = ['type'];
```

Comment:
what is the purpose of having `this.filterBy` as an array? Seems to hold always only on object?

Reply:
It's what bootstrap-table expects. Potentially you could filter by multiple fields (which we don't do in our case)